corehq/apps/es/REINDEX_PROCESS.md

```diff
@@ -0,0 +1,366 @@
+        ```
+
+4. At this step we should have a secondary index for every primary index. Writes will continue to happen to both primary and secondary indices simultaneously. Reads would still happening from the primary index.
+
+5. To switch the reads from primary index to the new secondary index, we need to swap the indices.
+To swap the indexes we need to follow the following steps:
+
+    1. Stop the pillows
+
+        ```
+        cchq <env> service pillowtop stop
+        ```
+    2. Copy the checkpoint ids for the pillows that depend on the index you are about to swap. For every index_cname run:
+
+        ```
+        cchq <env> django-manage  elastic_sync_multiplexed copy_checkpoints <index_cname>
+        ```
+    3. We will now swap the indexes in settings to  make the primary index as secondary and vice versa.
+    Update `environments/<env>/public.yml`, set
+        ```
+        ES_APPS_INDEX_SWAPPED = True
+        ES_CASE_SEARCH_INDEX_SWAPPED = True
+        ES_CASES_INDEX_SWAPPED = True
+        ES_DOMAINS_INDEX_SWAPPED = True
+        ES_FORMS_INDEX_SWAPPED = True
+        ES_GROUPS_INDEX_SWAPPED = True
+        ES_SMS_INDEX_SWAPPED = True
+        ES_USERS_INDEX_SWAPPED = True
+        ```
+    4. From the control machine, run update-config and restart commcare services so that the new settings are applied. This will also restart the pillows that were stopped in the previous step.
+        ```
+        cchq <env> update-config
+        cchq <env> service commcare restart
+        ```
+        After this step, the indexes will be swapped and reads will now happen from the newly created secondary indices.
+
+    5. Look around in CommcareHQ, test out things dealing with ES like Submit a form and ensure that it comes up in reports. If you have metrics setup for pillows, verify that the change feed is looking good.
+
+    6. It is recommended to keep the indices in this state for at least 2 working days. This will provide a safe window to fall back if needed.
+
+6. When you are confident that things are working fine with the new index, you are all set to turn off the multiplexer settings. Update `environments/<env>/public.yml` with following values:
+    ```
+    ES_APPS_INDEX_MULTIPLEXED = False
+    ES_CASE_SEARCH_INDEX_MULTIPLEXED = False
+    ES_CASES_INDEX_MULTIPLEXED = False
+    ES_DOMAINS_INDEX_MULTIPLEXED = False
+    ES_FORMS_INDEX_MULTIPLEXED = False
+    ES_GROUPS_INDEX_MULTIPLEXED = False
+    ES_SMS_INDEX_MULTIPLEXED = False
+    ES_USERS_INDEX_MULTIPLEXED = False
+    ```
+
+    From control machine, run update-config and restart commcare services so that new settings are applied.
+
+    ```
+    cchq <env> update-config
+    cchq <env> service commcare restart
+    ```
+
+7. At this stage, writes have been stopped on primary indices. The older indices are eligible for deletion. It is recommended to wait for 6 hours after the multiplexer is turned off. The older indices can be deleted by running.
```

Comment:
> At this stage, writes have been stopped on primary indices

Writes are stopped on the primary indices or secondary indices? I thought at this point we would have swapped the indices to have the older indices be the secondary indices?

Reply:
> I thought at this point we would have swapped the indices to have the older indices be the secondary indices?

This is correct. I think confusion here is on what I am considering as primary index, even after swapping I am not changing the names. 
I think the calling them older index and newer index would be much more appropriate here.